File: cpp/program/Elektronik.cpp
#include <iostream>
#include "Toko.cpp"
using namespace std;

// Class Elektronik (Child) dari class Toko (Parent)
class Elektronik : public Toko {
private:
    string kategori; // Atribut kategori
    int jumlahStok; // Atribut jumlah Stok
    string statusBarang; // Atribut status Barang

public:
    // Constructor kosong
    Elektronik() {
    }

    // Constructor dengan parameter
    Elektronik(string id, string namaProduk, int harga,
               string kategori, int jumlahStok, string statusBarang)
        : Toko(id, namaProduk, harga) {
        this->kategori = kategori;
        this->jumlahStok = jumlahStok;
        this->statusBarang = statusBarang;
    }

    // Getter setter
    string getKategori() const {
        return kategori;
    }
    void setKategori(string kategori) {
        this->kategori = kategori;
    }

    int getJumlahStok() const {
        return jumlahStok;
    }
    void setJumlahStok(int jumlahStok) {
        this->jumlahStok = jumlahStok;
    }

    string getStatusBarang() const {
        return statusBarang;
    }
    void setStatusBarang(string statusBarang) {
        this->statusBarang = statusBarang;
    }
};

File: java/program/Elektronik.java
public class Elektronik extends Toko {
    private String kategori;
    private int jumlahStok;
    private String statusBarang;

    // Constructor default
    public Elektronik() {
        super();
        this.kategori = "";
        this.jumlahStok = 0;
        this.statusBarang = "";
    }

    // Constructor dengan parameter
    public Elektronik(String id, String namaProduk, int harga,
                      String kategori, int jumlahStok, String statusBarang) {
        super(id, namaProduk, harga); // panggil constructor parent
        this.kategori = kategori;
        this.jumlahStok = jumlahStok;
        this.statusBarang = statusBarang;
    }

    // Getter & Setter
    public String getKategori() {
        return kategori;
    }

    public void setKategori(String kategori) {
        this.kategori = kategori;
    }

    public int getJumlahStok() {
        return jumlahStok;
    }

    public void setJumlahStok(int jumlahStok) {
        this.jumlahStok = jumlahStok;
    }

    public String getStatusBarang() {
        return statusBarang;
    }

    public void setStatusBarang(String statusBarang) {
        this.statusBarang = statusBarang;
    }
}
